### PR TITLE
Added MobFox Adapter

### PR DESCRIPTION
**Pull request by francoroy**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change

- test parameters for validating bids ( for description visit http://docs.mobfox.com/docs/banners )
```
{
  bidder: 'mobfox',
  params: {
    s: "267d72ac3f77a3f447b32cf7ebf20673" // required - The hash of your inventory to identify which app is making the request,
    imp_instl: 1 // optional - set to 1 if using interstitial otherwise delete or set to 0
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

### CODE REVIEW

File: src/adapters/mobfox.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  }
+
+  function sendBidRequest(bid) {
+    let requestParams = buildBidRequest(bid);
+    let queryString = buildQueryStringFromParams(requestParams);
+
+    ajax.ajax(`${BID_REQUEST_BASE_URL}?${queryString}`, {
+      success(resp, xhr) {
+        if (xhr.getResponseHeader("Content-Type") == "application/json")
+          resp = JSON.parse(resp);
```
Comment (jaiminpanchal27) [inline src/adapters/mobfox.js]: wrap all JSON.parse in try catch block

**Comment by francoroy** [comment]
Hi, any progress?
Thanks

**Comment by jaiminpanchal27** [comment]
@francoroy Changes looks good. But not able to validate bids. 
I am using [hello-wold](https://github.com/prebid/Prebid.js/blob/master/integrationExamples/gpt/hello_world.html) test page

**Comment by francoroy** [comment]
@jaiminpanchal27 Thanks. I can share my hello world page:
```
var adUnits = [{
        code: 'div-gpt-ad-1460505748561-0',
        sizes: [[320, 480]],

        // Replace this object to test a new Adapter!
        bids: [{
          bidder: 'mobfox',
          params: {
            s: '267d72ac3f77a3f447b32cf7ebf20673',
            imp_instl: 1
          }
        }]

      }];
```

**Comment by jaiminpanchal27** [comment]
@francoroy Got it. Thanks. 
Now able to see ad with size 320x480.

**Comment by dbemiller** [comment]
I did find a bit of an issue here...

If your server sends back a "no bid available" error, the adapter is still trying to add the bid to the bidmanager.

This was likely working when you wrote it, but changes in #1277 (in particular, [this line](https://github.com/prebid/Prebid.js/pull/1277/files#diff-96cd0debf7b766d72d2bcf987385e960L95)) cause it to break.

Since that was an unintended change, I'm replacing the old behavior in #1372... but I don't think this behavior is ideal. If an adapter doesn't have a bid to add, then it shouldn't be calling the `addBidResponse` method.

**Comment by francoroy** [comment]
@dbemiller
"Since my changes were unintended, I'm replacing the old behavior in #1372... but I don't think this is ideal. If an adapter doesn't have a bid to add, then it shouldn't be calling the addBidResponse method"
Hi! so I don't understand, should I or should I not call addBidResponse when no bid is available?

**Comment by dbemiller** [comment]
There's a `NO_BID` bid. For example: https://github.com/prebid/Prebid.js/blob/master/modules/appnexusAstBidAdapter.js#L175

**Comment by francoroy** [comment]
@dbemiller this is what I already do in my onBidResponseError function:

  let bidResponse = bidfactory.createBid(CONSTANTS.STATUS.NO_BID, bid);
  bidResponse.bidderCode = BIDDER_CODE;
  bidmanager.addBidResponse(bid.placementCode, bidResponse);


Is it ok then?

**Comment by dbemiller** [comment]
You're calling addBidResponse twice, and the second call uses an undefined bid. See the attached screenshot & test page.

screen shot 2017-07-20 at 4 32 45 pm

```
<!--

  This page calls a single bidder for a single ad slot. It can be considered a "hello world" example for using
  Prebid with the Google Publisher Tag.

  It also makes a good test page for new adapter PR submissions. Simply set your server's Bid Params object in the
  bids array inside the adUnits, and it will use your adapter to load an ad.

  NOTE that many ad servers won't send back an ad if the URL is localhost... so you might need to
  set an alias in your /etc/hosts file so that you can load this page from a different domain.

-->

<html>
<head>
    <script>
      var PREBID_TIMEOUT = 700;

      var adUnits = [{
        code: 'div-gpt-ad-1460505748561-0',
        sizes: [[300, 250], [300,600]],

        // Replace this object to test a new Adapter!
        bids: [{
          bidder: 'mobfox',
          params: {
            s: "267d72ac3f77a3f447b32cf7ebf20673", // required - The hash of your inventory to identify which app is making the request,
            imp_instl: 1 // optional - set to 1 if using interstitial otherwise delete or set to 0
          }
        }]
      }];

      var pbjs = pbjs || {};
      pbjs.que = pbjs.que || [];

    </script>

    <script type="text/javascript" src="../../build/dev/prebid.js" async></script>
    <script>
      var googletag = googletag || {};
      googletag.cmd = googletag.cmd || [];
      googletag.cmd.push(function() {
        googletag.pubads().disableInitialLoad();
      });

      pbjs.que.push(function() {
        pbjs.addAdUnits(adUnits);
        pbjs.requestBids({
          bidsBackHandler: sendAdserverRequest
        });
      });

      function sendAdserverRequest() {
        if (pbjs.adserverRequestSent) return;
        pbjs.adserverRequestSent = true;
        googletag.cmd.push(function() {
          pbjs.que.push(function() {
            pbjs.setTargetingForGPTAsync();
            googletag.pubads().refresh();
          });
        });
      }

      setTimeout(function() {
        sendAdserverRequest();
      }, PREBID_TIMEOUT);

    </script>

    <script>
      (function () {
        var gads = document.createElement('script');
        gads.async = true;
        gads.type = 'text/javascript';
        var useSSL = 'https:' == document.location.protocol;
        gads.src = (useSSL ? 'https:' : 'http:') +
          '//www.googletagservices.com/tag/js/gpt.js';
        var node = document.getElementsByTagName('script')[0];
        node.parentNode.insertBefore(gads, node);
      })();
    </script>

    <script>
      googletag.cmd.push(function () {
        googletag.defineSlot('/19968336/header-bid-tag-0', [[300, 250], [300, 600]], 'div-gpt-ad-1460505748561-0').addService(googletag.pubads());

        googletag.pubads().enableSingleRequest();
        googletag.enableServices();
      });
    </script>
</head>

<body>
<h2>Prebid.js Test</h2>
<h5>Div-1</h5>
<div id='div-gpt-ad-1460505748561-0'>
    <script type='text/javascript'>
      googletag.cmd.push(function() { googletag.display('div-gpt-ad-1460505748561-0'); });
    </script>
</div>
</body>
</html>
```

**Comment by francoroy** [comment]
@dbemiller 
Hi, thanks for the review. 
I didn't manage to reproduce the issue with your test page but I found the bug and fixed it  (onBidResponseError was called twice on invalid json).

Thanks

**Comment by dbemiller** [comment]
@francoroy I'm still getting the exact same error. I've attached a screenshot from the same spot, but looking at your code.

screen shot 2017-07-24 at 12 14 54 pm

Your server is responding with a 200, but the response is `{"error": "no Ad Available"}`. `transformResponse` returns undefined, and `onBidResponse` tries to add "undefined" as a bid response.

It reproduces in your unit tests, too:

```
 let bidResponse1 = stubAddBidResponse.getCall(0).args[1];
 expect(stubAddBidResponse.calledTwice).to.equal(true); // This passes, but addBidResponse should only be called once
expect(bidPlacementCode1).to.equal('test-gpt-div-1234');
```

**Comment by francoroy** [comment]
@dbemiller Thanks, I've fixed it

**Pull request merged**